atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweepTest.java

```diff
@@ -0,0 +1,273 @@
+ */
+
+package com.palantir.atlasdb.sweep.queue;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Matchers.anyList;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.List;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.palantir.atlasdb.AtlasDbConstants;
+import com.palantir.atlasdb.AtlasDbTestCase;
+import com.palantir.atlasdb.encoding.PtBytes;
+import com.palantir.atlasdb.keyvalue.api.Cell;
+import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.keyvalue.api.Value;
+import com.palantir.atlasdb.keyvalue.api.WriteReference;
+import com.palantir.atlasdb.protos.generated.TableMetadataPersistence;
+import com.palantir.atlasdb.table.description.ColumnMetadataDescription;
+import com.palantir.atlasdb.table.description.NameMetadataDescription;
+import com.palantir.atlasdb.table.description.TableMetadata;
+import com.palantir.atlasdb.transaction.api.ConflictHandler;
+import com.palantir.atlasdb.transaction.api.PreCommitCondition;
+import com.palantir.atlasdb.transaction.api.Transaction;
+import com.palantir.atlasdb.transaction.api.TransactionConflictException;
+
+public class TargetedSweepTest extends AtlasDbTestCase {
+    private static final TableReference TABLE_CONS = TableReference.createFromFullyQualifiedName("test.1");
+    private static final TableReference TABLE_THOR = TableReference.createFromFullyQualifiedName("test.2");
+    private static final Cell TEST_CELL = Cell.create(PtBytes.toBytes("row1"), PtBytes.toBytes("col1"));
+    private static final byte[] TEST_DATA = new byte[]{1};
+    private static final WriteReference SINGLE_WRITE = WriteReference.write(TABLE_CONS, TEST_CELL);
+    private static final WriteReference SINGLE_DELETE = WriteReference.tombstone(TABLE_CONS, TEST_CELL);
+
+    @Override
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+        keyValueService.createTable(TABLE_CONS, AtlasDbConstants.GENERIC_TABLE_METADATA);
+        keyValueService.createTable(TABLE_THOR, new TableMetadata(new NameMetadataDescription(),
+                new ColumnMetadataDescription(),
+                ConflictHandler.RETRY_ON_WRITE_WRITE,
+                TableMetadataPersistence.CachePriority.WARM,
+                false,
+                0,
+                false,
+                TableMetadataPersistence.SweepStrategy.THOROUGH,
+                false,
+                TableMetadataPersistence.LogSafety.UNSAFE)
+                .persistToBytes());
```

Comment:
This might be easier:

```
        keyValueService.createTable(TABLE_THOR, TableMetadataPersistence.TableMetadata.newBuilder(
                TableMetadataPersistence.TableMetadata.parseFrom(AtlasDbConstants.GENERIC_TABLE_METADATA))
                .setSweepStrategy(TableMetadataPersistence.SweepStrategy.THOROUGH)
                .build()
                .toByteArray());
```